src/python/pants/rules/core/generate_pants_ini.py

```diff
@@ -16,36 +16,50 @@
 from pants.engine.goal import Goal, GoalSubsystem
 from pants.engine.rules import goal_rule
 from pants.engine.selectors import Get
-from pants.version import VERSION as pants_version
+from pants.version import VERSION
 
 
 class GeneratePantsIniOptions(GoalSubsystem):
   """Generate pants.ini with sensible defaults."""
   name = 'generate-pants-ini'
 
+  @classmethod
+  def register_options(cls, register) -> None:
+    super().register_options(register)
+    register('--v2-only', type=bool, fingerprint=True, default=False,
+             help='Support only v2 rules. If unspecified, this repo will also run v1 tasks.')
+
 
 class GeneratePantsIni(Goal):
   subsystem_cls = GeneratePantsIniOptions
 
 
 @goal_rule
-async def generate_pants_ini(console: Console, workspace: Workspace) -> GeneratePantsIni:
+async def generate_pants_ini(console: Console, workspace: Workspace,
+                             options: GeneratePantsIniOptions) -> GeneratePantsIni:
   pants_ini_content = dedent(f"""\
     [GLOBAL]
-    pants_version: {pants_version}
+    pants_version: {VERSION}
+    v1: {'False' if options.values.v2_only else 'True'}
```

Comment:
FYI: no need to quote the `False` and `True` values. `{}` in f-strings will call `format()` on the underlying object (which calls `__format__()`).

Reply:
I am aware, but this feels like it communicates intent more clearly.